### PR TITLE
Helpful fungal analysis scripts

### PR DESCRIPTION
**Pull request by tives82**
This pull request includes the set of Python and shell scripts that I use in C. auris analysis as well as updates to the README.md. They significantly streamline the process of handling and analyzing C. auris WGS data and significantly cut down the time spent manually handling data. 

The scripts include:

merge_c_auris_LIMS_export_files.py: Automates the merging of daily C. auris LIMS export reports, simplifying the process of gathering needed sample metadata for WGS analysis. This script reads, filters, and combines data from two CSV reports, outputting a consolidated Excel file.

healthcare_facility_of_origin_city.py: Aims to associate each C. auris sample with its originating healthcare facility's city. This script processes and merges data from key input files to fill gaps in city information.

gather_vcfs.sh: Designed to speed up the collection of VCF files necessary for mycoSNP analysis. It copies vcf files from mycosnp_results directories and utilizes GNU Parallel for parallel processing.

changeseqids.py: Addresses the need for standardized sequence identifiers in mycoSNP analysis, particularly analysis requests and REDCap submissions. This script replaces sequence IDs in the 'vcf-to-fasta.fasta' file with alternate IDs that comply with the naming conventions set by the Mycotic Diseases Branch at the CDC.

### CODE REVIEW

File: gather_vcfs.sh [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        echo "Successfully copied VCF for sample: $sample_id"
+    fi
+}
+
+# Export the function for parallel to access
+export -f copy_files
+
+# Use parallel to invoke the function for each line in the file
+cat samples.txt | parallel --colsep "\t" copy_files {1} {2}
+
```
Comment (jwarnn) [inline gather_vcfs.sh:40]: I was able to get this script to run; but the first time it ran I had an extra line at the bottom which created made the script copy all vcf and tbi files about 4,000. I think you should put some code in to catch such an occurrence. 
```
grep . samples.txt  > samples.txt
```
That code would eliminate those extra lines. Or the user can be more careful.

File: healthcare_facility_of_origin_city.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import sys  # system-specific parameters and functions
+
+"""
+Author: Tom Iverson
+Released: 2023-11-30
+Version: 1.0.0
+Description:
+This script's main functionality is to determine a samples' city from the C. auris LIMS export data (ARLN_Specimen_ID, Healthcare_facility_of_origin_name, and Healthcare_facility_of_origin_state).
+It does this by taking a 'samples.txt' file that the user has created with the following headers "ARLN_Specimen_ID, Healthcare_facility_of_origin_name, Healthcare_facility_of_origin_state" and merges
+the data with 'C_auris_Healthcare_Facility_of_origin_name_city_state.txt'.
```
Comment (jwarnn) [inline healthcare_facility_of_origin_city.py:12]: C_auris_Healthcare_Facility_of_origin_name_city_state.txt is this a file that you manually maintain? How do you find the locations of the facilities? Just through google?

**Review by jwarnn — APPROVED** [review]
I was able to get all four scripts to run. Because of user error one of the scripts did not work as intended and I suggest adding a stoping mechanism to prevent this behavior or I suggested some code. I also had a question for healthcare_facility_of_origin_city.py and the answer might be useful in the README.md. I am going to approve this PR even with my suggestion because they are fairly marginal concerns.

**Comment by jwarnn** [comment]
I should also say that I am sorry my review took so long. It must have slipped off my to do list and I forgot about it.

File: gather_vcfs.sh (hunk 1 above)
Comment (tives82) [reply gather_vcfs.sh:40]: Added the following to fix this:

Filter out any empty lines from samples.txt and overwrite it
grep . samples.txt > temp.txt && mv temp.txt samples.txt

File: healthcare_facility_of_origin_city.py (hunk 2 above)
Comment (tives82) [reply healthcare_facility_of_origin_city.py:12]: Clarified the readme to answer these questions.

**Pull request merged**